ast/compile.go

```diff
@@ -861,6 +876,24 @@ func checkUndefinedFuncs(x interface{}, arity func(Ref) int, rwVars map[Var]Var)
 	return errs
 }
 
+func arityMismatchBuiltin(env *TypeEnv, f Ref, t *types.Function, loc *location.Location, operands []*Term) *Error {
```

Comment:
nit: we have some functions in check.go that generate type errors for cases like these (e.g., https://github.com/open-policy-agent/opa/blob/main/ast/check.go#L1011) could we re-use them here or refactor so that we have one common set of functions for generating type errors? Trying to cut back on duplication so that we can ensure consistency.

Reply:
👍 Updated.